web3/middleware/filter.py

```diff
@@ -0,0 +1,131 @@
+import itertools
+
+from eth_utils import (
+    to_hex,
+)
+from toolz.itertoolz import (
+    concat,
+)
+
+
+class RangeCounter:
+    # ie (0, 4),(5, 9),(...)
+    def __init__(self, start, stop=None, step=5):
+        self.start = start
+        self.stop = stop
+        self.step = step
+        self._count_from = itertools.count(
+            start=self.start,
+            step=self.step)
+
+    def __iter__(self):
+        return self
+
+    def __next__(self):
+        for _start in self._count_from:
+            if self.stop and _start > self.stop:
+                raise StopIteration
+            _end = _start + (self.step - 1)
+            if self.stop and self.stop < _end:
+                _end = self.stop
+            return _start, _end
+
+
+def unload_latest_blocks(web3, fromBlock, toBlock):
```

Comment:
We've maintained using `fromBlock` instead of `from_block` in most public APIs, but for internal APIs I'd prefer we stick with `from_block` style casing.